test/quantization/test_quantize.py

```diff
@@ -940,40 +940,64 @@ def checkQuantized(model):
     def test_quantized_rnn(self, qconfig, dtype):
         r"""Test dynamic quantization, scriptability and serialization for dynamic quantized lstm modules on int8 and fp16
         """
-        model = RNNDynamicModel('LSTM').eval()
         niter = 10
         x = torch.tensor([[100, -155],
                           [-155, 100],
                           [100, -155]], dtype=torch.float).unsqueeze(0).repeat(niter, 1, 1)
         qconfig_dict = {
-            torch.nn.LSTM : qconfig
+            torch.nn.LSTM : qconfig,
+            torch.nn.GRU: qconfig
         }
-        if dtype == torch.float16:
-            model_quantized = quantize_dynamic(model=model, dtype=dtype)
-        else:
-            model_quantized = quantize_dynamic(model=model, qconfig_spec=qconfig_dict, dtype=dtype)
-
-        # Smoke test extra reprs
-        self.assertTrue('DynamicQuantizedLSTM' in str(model_quantized))
-        self.checkDynamicQuantizedModule(model_quantized.mod, torch.nn.quantized.dynamic.LSTM, dtype)
-        self.checkScriptable(model_quantized, [[x]], check_save_load=True)
-
-        class ScriptWrapperPacked(torch.nn.Module):
-            def __init__(self, cell):
-                super(ScriptWrapperPacked, self).__init__()
-                self.cell = cell
-
-            def forward(self,
-                        x  # type: PackedSequence
-                        ):
-                # type: (...) -> Tuple[PackedSequence, Tuple[torch.Tensor, torch.Tensor]]
-                return self.cell(x)
-
-        packed_input = torch.nn.utils.rnn.pack_padded_sequence(x, torch.tensor([10, 5, 2]))
-        model_with_packed_input = ScriptWrapperPacked(model_quantized.mod)
-        scripted = torch.jit.script(model_with_packed_input)
-        # We cannot trace with input dtype being a packed sequence
-        self._checkScriptable(model_with_packed_input, scripted, [[packed_input]], True)
+
+        for module_type in ['LSTM', 'GRU']:
```

Comment:
should we have a separate test case for each?

Reply:
Sure, can refactor the tests with some common code